webhook/validator/validator.go

```diff
@@ -0,0 +1,82 @@
+package validator
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	dfv1 "github.com/numaproj/numaflow/pkg/apis/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/client/clientset/versioned/typed/numaflow/v1alpha1"
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+type Validator interface {
+	ValidateCreate(context.Context) *admissionv1.AdmissionResponse
+	ValidateUpdate(context.Context) *admissionv1.AdmissionResponse
+}
+
+// GetValidator returns a Validator instance
+func GetValidator(ctx context.Context, client kubernetes.Interface, ISBSVCClient v1alpha1.InterStepBufferServiceInterface, PipelineClient v1alpha1.PipelineInterface, kind metav1.GroupVersionKind, oldBytes []byte, newBytes []byte) (Validator, error) {
+	log := logging.FromContext(ctx)
+	switch kind.Kind {
```

Comment:
Can we use `switch kind {` and use the constants in https://github.com/numaproj/numaflow/blob/main/pkg/apis/numaflow/v1alpha1/register.go for `case`?

Reply:
There is a type mismatch here with conflicting packages `metav1` and `schema`:

```
invalid case dfv1.ISBGroupVersionKind in switch on kind (mismatched types schema.GroupVersionKind and "k8s.io/apimachinery/pkg/apis/meta/v1".GroupVersionKind)
```

Reply:
If we still use `kind.Kind` for switch, we can check each `case` like this by comparing just the string `Kind`:
```
	switch kind.Kind {
	case dfv1.ISBGroupVersionKind.Kind:
```

Reply:
> 

This is good.